src/EnsembleKalmanProcess.jl

```diff
@@ -336,4 +337,324 @@ function update_ensemble!(ekp::EnsembleKalmanProcess{FT, IT, Sampler{FT}}, g_in:
+    
+    
+end
+
+
+"""
+construct_sigma_ensemble
+Construct the sigma ensemble, based on the mean x_mean, and covariance x_cov
+"""
+function construct_sigma_ensemble(uki::EnsembleKalmanProcess{FT, IT, Unscented}, x_mean::Array{FT}, x_cov::Array{FT,2}) where {FT<:AbstractFloat, IT<:Int}
+    N_ens = uki.N_ens
+    N_x = size(x_mean,1)
+    @assert(N_ens == 2*N_x+1)
+    
+    c_weights = uki.process.c_weights
+    
+    
+    chol_xx_cov = cholesky(Hermitian(x_cov)).L
+    
+    x = zeros(FT, N_x, 2*N_x+1)
+    x[:, 1] = x_mean
+    for i = 1: N_x
+        x[:, i+1] = x_mean + c_weights[i]*chol_xx_cov[:,i]
+        x[:, i+1+N_x] = x_mean - c_weights[i]*chol_xx_cov[:,i]
+    end
+    
+    return x
+end
+
+
+"""
+construct_mean x_mean from ensemble x
+"""
+function construct_mean(uki::EnsembleKalmanProcess{FT, IT,Unscented}, x::Array{FT,2}) where {FT<:AbstractFloat, IT<:Int}
+    N_x, N_ens = size(x)
+    
+    @assert(uki.N_ens == N_ens)
+    
+    x_mean = zeros(FT, N_x)
+    
+    mean_weights = uki.process.mean_weights
+    
+    
+    for i = 1: N_ens
+        x_mean += mean_weights[i]*x[:, i]
+    end
+    
+    return x_mean
+end
+
+"""
+construct_cov xx_cov from ensemble x and mean x_mean
+"""
+function construct_cov(uki::EnsembleKalmanProcess{FT, IT,Unscented}, x::Array{FT,2}, x_mean::Array{FT}) where {FT<:AbstractFloat, IT<:Int}
+    N_ens, N_x = uki.N_ens, size(x_mean,1)
+    
+    cov_weights = uki.process.cov_weights
+    
+    xx_cov = zeros(FT, N_x, N_x)
+    
+    for i = 1: N_ens
+        xx_cov .+= cov_weights[i]*(x[:,i] - x_mean)*(x[:,i] - x_mean)'
+    end
+    
+    return xx_cov
+end
+
+"""
+construct_cov xy_cov from ensemble x and mean x_mean, ensemble obs_mean and mean y_mean
+"""
+function construct_cov(uki::EnsembleKalmanProcess{FT, IT, Unscented}, x::Array{FT,2}, x_mean::Array{FT}, obs_mean::Array{FT,2}, y_mean::Array{FT}) where {FT<:AbstractFloat, IT<:Int, P<:Process}
+    N_ens, N_x, N_y = uki.N_ens, size(x_mean,1), size(y_mean,1)
+    
+    cov_weights = uki.process.cov_weights
+    
+    xy_cov = zeros(FT, N_x, N_y)
+    
+    for i = 1: N_ens
+        xy_cov .+= cov_weights[i]*(x[:, i] - x_mean)*(obs_mean[:, i] - y_mean)'
+    end
+    
+    return xy_cov
+end
+
+"""
+uki prediction step : generate sigma points
+"""
+function update_ensemble_prediction!(uki::EnsembleKalmanProcess{FT, IT,Unscented}) where {FT<:AbstractFloat, IT<:Int}
+    
+    uki.process.iter += 1
+    # update evolution covariance matrix
+    if uki.process.update_freq > 0 && uki.process.iter%uki.process.update_freq == 0
+        uki.process.Σ_ω = (2 - uki.process.α_reg^2)uki.process.uu_cov[end]
+    end
+    
+    u_mean  = uki.process.u_mean[end]
+    uu_cov = uki.process.uu_cov[end]
+    
+    α_reg = uki.process.α_reg
+    r = uki.process.r
+    Σ_ω = uki.process.Σ_ω
+    
+    N_u, N_ens = length(uki.process.u_mean[1]),  uki.N_ens
```

Comment:
By the looks of it, we don't even use `N_ens` in this function. so this can be removed

Reply:
updated. Thanks